numpy/core/tests/test_umath.py

```diff
@@ -357,20 +402,47 @@ def test_float_remainder_corner_cases(self):
         # Check nans, inf
         with suppress_warnings() as sup:
             sup.filter(RuntimeWarning, "invalid value encountered in remainder")
+            sup.filter(RuntimeWarning, "invalid value encountered in divmod")
+            sup.filter(RuntimeWarning, "invalid value encountered in fmod")
+            sup.filter(RuntimeWarning, "divide by zero encountered in divmod")
             for dt in np.typecodes['Float']:
                 fone = np.array(1.0, dtype=dt)
                 fzer = np.array(0.0, dtype=dt)
                 finf = np.array(np.inf, dtype=dt)
                 fnan = np.array(np.nan, dtype=dt)
                 rem = np.remainder(fone, fzer)
                 assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
+                div, rem = np.divmod(fone, fzer)
+                assert_(np.isinf(div), 'dt: %s, div: %s' % (dt, rem))
+                assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
+                div, rem = np.divmod(fzer, fzer)
+                assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
+                assert_(np.isnan(div), 'dt: %s, rem: %s' % (dt, rem))
+                div, rem = np.divmod(finf, finf)
+                assert_(np.isnan(div), 'dt: %s, rem: %s' % (dt, rem))
+                assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
+                div, rem = np.divmod(finf, fzer)
+                assert_(np.isinf(div), 'dt: %s, rem: %s' % (dt, rem))
+                assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
                 # MSVC 2008 returns NaN here, so disable the check.
                 #rem = np.remainder(fone, finf)
                 #assert_(rem == fone, 'dt: %s, rem: %s' % (dt, rem))
                 rem = np.remainder(fone, fnan)
+                fmod = np.fmod(fone, fnan)
                 assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
+                assert_(np.isnan(fmod), 'dt: %s, fmod: %s' % (dt, fmod))
                 rem = np.remainder(finf, fone)
+                fmod = np.fmod(finf, fone)
+                assert_(np.isnan(fmod), 'dt: %s, fmod: %s' % (dt, fmod))
+                assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
+                rem = np.remainder(finf, finf)
+                fmod = np.fmod(finf, fone)
+                assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
+                assert_(np.isnan(fmod), 'dt: %s, fmod: %s' % (dt, fmod))
+                rem = np.remainder(finf, fzer)
+                fmod = np.fmod(finf, fzer)
                 assert_(np.isnan(rem), 'dt: %s, rem: %s' % (dt, rem))
+                assert_(np.isnan(fmod), 'dt: %s, fmod: %s' % (dt, fmod))
```

Comment:
Can just keep it, but in the future, I personally would go with the a simple `assert`. This was useful in nose, but pytest is already overly verbose anyway, the message adds nothing.

Reply:
sure, noted!